core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -163,21 +181,24 @@ private[spark] class SparkArithmeticException(
 private[spark] class SparkUnsupportedOperationException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String])
+    messageParameters: Map[String, String])
   extends UnsupportedOperationException(
     SparkThrowableHelper.getMessage(errorClass, errorSubClass.orNull, messageParameters))
-    with SparkThrowable {
+  with SparkThrowable {
 
   def this(
       errorClass: String,
       errorSubClass: String,
-      messageParameters: Array[String]) =
+      messageParameters: Map[String, String]) =
     this(
       errorClass = errorClass,
       errorSubClass = Some(errorSubClass),
       messageParameters = messageParameters)
 
-  override def getMessageParameters: Array[String] = messageParameters
+  override def getMessageParameters: Array[String] = {
```

Comment:
It is tested by QueryExecutionErrorsSuite "UNSUPPORTED_FEATURE: unsupported pivot operations"